### PR TITLE
Keras2

### PR DESCRIPTION
**Pull request by wsmoses**

### CODE REVIEW

**Comment by wsmoses** [comment]
@ftynse I think I finally got it.

..... the horrors....

File: builddeps/requirements_lock_3_10.txt
Comment (ftynse) [inline builddeps/requirements_lock_3_10.txt:1]: Is this needed? We seem to set python version to 3.12.

File: test/jaxmd.py [hunk 1]
```diff
@@ -1,55 +1,26 @@
...
         from jax_md import space
         from jax_md import energy
         from jax_md import simulate
         from jax_md import quantity
         from jax_md import partition
 
         lattice_constant = 1.37820
-        N_rep = 40
+        # We hit a GPU memory limit for N_rep = 40
+        N_rep = 10 if jax.default_backend() == "gpu" else 40
```
Comment (ftynse) [inline test/jaxmd.py:23]: Should we just keep 10 so we don't accidentally compare mismatching CPU/GPU later?

**Review by ftynse — APPROVED** [review]
I'll give it a try later on a clean machine to check that we have not hardcoded for your environment :)

File: test/test_utils.py [hunk 2]
```diff
@@ -1,43 +1,357 @@
...
+        "NVARCH",
+        "NV_CUDA_NSIGHT_COMPUTE_VERSION",
+        "NV_LIBNCCL_PACKAGE_NAME",
+        "NV_LIBNCCL_PACKAGE_VERSION",
+    ]:
+        os.environ.pop(nm, None)
+
+    runfiles = os.path.dirname(
+        os.path.dirname(os.path.dirname(os.path.abspath(__file__)))
+    )
```
[the remark this answers is not in the document]
Comment (wsmoses) [reply test/test_utils.py:30]: bazel does weird things faking paths

**Pull request closed without merging**